neuralop/layers/segment_csr.py

```diff
@@ -54,38 +60,23 @@ def segment_csr(
         # if batched, shape [b, n_reps, channels]
         # otherwise shape [n_reps, channels]
         if src.ndim == 3:
-            batched = True
             point_dim = 1
         else:
-            batched = False
             point_dim = 0
 
         # if batched, shape [b, n_out, channels]
         # otherwise shape [n_out, channels]
         output_shape = list(src.shape)
-        n_out = indptr.shape[point_dim] - 1
+        n_out = indptr.shape[0] - 1
+        n_reps = indptr[1:] - indptr[:-1]
+
         output_shape[point_dim] = n_out
 
-        out = torch.zeros(output_shape, device=src.device)
+        indices = torch.arange(n_out).to(indptr.device).repeat_interleave(n_reps)
+        out = torch.zeros(output_shape, device=src.device).index_add_(point_dim, indices, src)
+        if reduce == 'mean':
+            # scale the outputs by number of reduced neighbors, add eps to avoid div by zero
+            scale = (1 / (n_reps + eps)).unsqueeze(0).T
+            out = out * scale    
```

Comment:
Seems odd to just have one case in the if. Isn't this the most efficient place to add quadrature? If we assume they are already in the features, wouldn't it be the same for the mean scaling factor?

Reply:
Without the if, the code implements `reduce=="sum"` by default.